include/mxnet/tuple.h

```diff
@@ -404,9 +452,11 @@ class TShape : public Tuple<dim_t> {
   }
   /*! \return total number of elements in the shape */
   inline size_t Size() const {
+    if (this->ndim() == -1) return 0;
```

Comment:
Should this throw an error instead?

Reply:
The hope is to easily identify any illegal use of undefined shapes. what do you think?

Reply:
Agree. It doesn't make sense to get a size value when shape is unknown. Will change that.